ci/src/main.rs

```diff
@@ -223,18 +233,35 @@ fn bump_version(krate: &Crate, crates: &[Rc<RefCell<Crate>>], patch: bool) {
 
     // Update the dependencies of this crate to point to the new version of
     // crates that we're bumping.
-    let dependencies = match new_manifest["dependencies"].as_table_mut() {
-        Some(d) => d,
-        None => return,
-    };
-    for (dep_name, dep) in dependencies.iter_mut() {
-        if crates.iter().any(|k| *k.borrow().name == *dep_name) {
-            let dep_version = bump(dep["version"].as_str().expect("dep version"), patch);
-            dep["version"] = toml_edit::value(dep_version.clone());
+    let dependencies = new_manifest["dependencies"].as_table_mut();
+    if let Some(dependencies) = dependencies {
+        for (dep_name, dep) in dependencies.iter_mut() {
+            if crates.iter().any(|k| *k.borrow().name == *dep_name) {
+                let dep_version = bump(dep["version"].as_str().expect("dep version"), patch);
+                dep["version"] = toml_edit::value(dep_version.clone());
+            }
         }
     }
 
-    fs::write(&krate.path, new_manifest.to_string()).expect("failed to write new manifest");
+    fs::write(&krate.manifest_path, new_manifest.to_string())
+        .expect("failed to write new manifest");
+
+    if let Some(changelog_path) = &krate.changelog_path {
+        if krate.should_bump {
+            let todays_date = chrono::Local::now().format("%m-%d-%Y");
+            let mut changelog =
+                fs::read_to_string(changelog_path).expect("failed to read changelog");
+            changelog = changelog.replace(
+                "## Unreleased",
+                &format!(
+                    "## Unreleased\n\n## {} - {}",
+                    bump(&krate.version, patch),
+                    todays_date
+                ),
+            );
+            fs::write(changelog_path, changelog).expect("failed to write changelog");
+        }
+    }
 }
```

Comment:
Does this happen even if we fail to publish because the crate has already been published previously?

That is to say, I wouldn't want this to continually insert the `## Unreleased` when we've done so for the previous one.

Do we need to check to see if the file contents contains `format!("## {} - ", bump(&krate.version, patch))`?

Reply:
Oh never mind, this only happens for the `bump` command.